components/org.wso2.carbon.identity.api.server.organization.user.invitation.management/org.wso2.carbon.identity.api.server.organization.user.invitation.management.v1/src/main/java/org/wso2/carbon/identity/api/server/organization/user/invitation/management/v1/impl/GuestsApiServiceImpl.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright (c) 2023, WSO2 LLC. (http://www.wso2.com).
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.api.server.organization.user.invitation.management.v1.impl;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.wso2.carbon.identity.api.server.organization.user.invitation.management.v1.GuestsApiService;
+import org.wso2.carbon.identity.api.server.organization.user.invitation.management.v1.core.GuestApiServiceCore;
+import org.wso2.carbon.identity.api.server.organization.user.invitation.management.v1.model.AcceptanceRequestBody;
+import org.wso2.carbon.identity.api.server.organization.user.invitation.management.v1.model.IntrospectRequestBody;
+import org.wso2.carbon.identity.api.server.organization.user.invitation.management.v1.model.IntrospectSuccessResponse;
+import org.wso2.carbon.identity.api.server.organization.user.invitation.management.v1.model.InvitationRequestBody;
+import org.wso2.carbon.identity.api.server.organization.user.invitation.management.v1.model.InvitationSuccessResponse;
+import org.wso2.carbon.identity.api.server.organization.user.invitation.management.v1.model.InvitationsListResponse;
+
+import javax.ws.rs.core.Response;
+
+/**
+ * Implementation
+ */
+public class GuestsApiServiceImpl implements GuestsApiService {
+
+    @Autowired
+    private GuestApiServiceCore guestApiServiceCore;
+
+    @Override
+    public Response invitationAcceptPost(AcceptanceRequestBody acceptanceRequestBody) {
+
+        guestApiServiceCore.acceptInvitation(acceptanceRequestBody);
+        return Response.noContent().build();
+    }
+
+    @Override
+    public Response invitationDelete(String invitationId) {
+
+        guestApiServiceCore.deleteInvitation(invitationId);
+        return Response.noContent().build();
+    }
+
+    @Override
+    public Response invitationIntrospectPost(IntrospectRequestBody introspectRequestBody) {
+
+        IntrospectSuccessResponse introspectSuccessResponse =
+                guestApiServiceCore.introspectInvitation(introspectRequestBody.getConfirmationCode());
+        return Response.ok().entity(introspectSuccessResponse).build();
+    }
+
+    @Override
+    public Response invitationListGet(String filter) {
+
+        InvitationsListResponse invitationsListResponse = guestApiServiceCore.getInvitations(filter);
```

Comment:
To avoid public methods signature changes and having multiple overloaded methods, shall we introduce pagination params as well and return not implemented error in this phase

Reply:
Done